File: src/main/java/jp/osak/haggledehaghag/viewmodel/PlayerView.java
package jp.osak.haggledehaghag.viewmodel;

import jp.osak.haggledehaghag.model.Player;

public record PlayerView(
        String displayName,
        String playerKey
) {
    public static PlayerView of(final Player player) {
        return new PlayerView(player.displayName(), player.playerKey());
    }
}

File: src/main/java/jp/osak/haggledehaghag/viewmodel/PlayerView.kt
package jp.osak.haggledehaghag.viewmodel

import jp.osak.haggledehaghag.model.Player

data class PlayerView(
        val displayName: String,
        val playerKey: String
) {
    constructor(player: Player) : this(player.displayName, player.playerKey)
}
